x-pack/plugins/fleet/public/applications/fleet/sections/settings/components/edit_output_flyout/output_health.tsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiCallOut } from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+import React, { useCallback, useEffect, useState } from 'react';
+
+import type { GetOutputHealthResponse } from '../../../../../../../common/types';
+
+import { sendGetOutputHealth, useStartServices } from '../../../../hooks';
+import type { Output } from '../../../../types';
+
+interface Props {
+  output: Output;
+}
+const REFRESH_INTERVAL_MS = 5000;
+
+export const OutputHealth: React.FunctionComponent<Props> = ({ output }) => {
+  const { notifications } = useStartServices();
+  const [outputHealth, setOutputHealth] = useState<GetOutputHealthResponse | null>();
+  const fetchData = useCallback(async () => {
+    try {
+      const response = await sendGetOutputHealth(output.id);
+      if (response.error) {
+        throw response.error;
+      }
+      setOutputHealth(response.data);
+    } catch (error) {
+      notifications.toasts.addError(error, {
+        title: i18n.translate('xpack.fleet.output.errorFetchingOutputHealth', {
+          defaultMessage: 'Error fetching output state',
+        }),
+      });
+    }
+  }, [output.id, notifications.toasts]);
+
+  // Send request to get output health
+  useEffect(() => {
+    fetchData();
+    const interval = setInterval(() => {
+      fetchData();
+    }, REFRESH_INTERVAL_MS);
+
+    return () => clearInterval(interval);
+  }, [fetchData]);
```

Comment:
This could be an interesting and fairly small chunk of code to refactor to `react-query`. A `useQuery` with a `refetchInterval` set will get the job done. Would probably eliminate a decent chunk of code if you're willing to explore the refactor. 

We'd also need to add a quick `sendRequestForRq` wrapper in place of the `sendGetOutputHealth` function. 

Obviously non-blocking, but it seems like a good opportunity to try out the polling behavior with RQ that we'd also want to use on the agent listing table and elsewhere down the road. What do you think?

Reply:
updated to use react query, indeed it is much shorter this way